Web_Development/Javascript/Topics/5.Syntax/2.Loops/README.md

```diff
@@ -0,0 +1,237 @@
+
+### 2. For In Loop:
+
+Syntax of ***for in*** loop :
+```js
+for(key in object){
+    //Block of code
+}
+```
+- *key* - You can imagine it as the index value of each element in an array. But in case of objects, we have key-value pairs associated with every entry in the object. Thus we can iterate over every entry of an object using it's keys.
+
+***For Example :***
+
+- For in over **objects**
+```js
+const student = {name:"Gurpreet", roll_no:"101", age:19};
+
+for (let key in student) {
+  console.log("Student "+key+" is "+student[key]);
+}
+```
+Output :
+```
+Student name is Gurpreet
+Student roll_no is 101
+Student age is 19
+```
+
+- For in over **arrays**
+```js
+const arr = [1,2,3,4,5];
+let sum = 0;
+
+for (let index in arr) {
+  sum += arr[index];
+}
+
+console.log("The array sum is : " + sum);
+```
+Output :
+```
+The array sum is : 15
+```
+
+>### ***Curious Corner :***
+>- Do not use ``for in`` over an array if the index **order** is important.
+>
+>- It is better to use a ``for loop``, a ``for of`` loop, or ``Array.forEach()`` when the order is important.
+
+### 3. For of Loop:
+Syntax of ***for in*** loop :
+```js
+for (value of iterable) {
+  // block of code
+}
+```
+- *value* - For every iteration the value of the next property is assigned to the variable
+- *iterable* - An object which can be iterated
+
+***For Example :***
+
+- For in over **arrays**
+```js
+const arr = ["Mango", "Orange", "Apple"];
+let fruits = "";
+
+for (let value of arr) {
+  fruits += value + " ";
+}
+
+console.log("My favorite fuits are : " + fruits);
+```
+Output :
+```
+My favorite fuits are : Mango Orange Apple
+```
+> **Note:** You can also use for of loop with *strings*
+
+### 4. While Loop:
+Syntax of ***while*** loop :
+```js
+while (condition) {
+  // code block to be executed
+}
+```
+- *condition* - Loop will execute until the condition has been falsified
+
+***For Example :***
+
+```js
+let i = 1;
+
+// Printing first 5 natural numbers
+while (i <= 5) {
+  console.log(i+" ");
+  i++;
+}
+```
+Output :
+```
+1 
+2 
+3
+4
+5
+```
+
+### 5. Do While Loop:
+Syntax of ***do while*** loop :
+```js
+do {
+  // block of code
+}
+while (condition);
+```
+- *condition* - Loop will execute until the condition has been falsified
+
+> **Note :** ``do while`` loop is different from ``while`` loop because ``do while`` loop checks the condition ***after*** the execution of each loop.
+
+***For Example :***
+
+```js
+let students = ["Gurpreet", "Kailash", "Anant", "Pratul"];
+let i=0;
+do
+{
+  console.log(students[i]);
+  i++;
+}
+while(i<4)
+```
+Output :
+```
+Gurpreet
+Kailash
+Anant
+Pratul
+```
+
+>### ***Curious Corner :***
+> You can iterate over an array which the help of ``Array.forEach()`` method as well instead of using loops. It was introduced with **ES5** version.
```

Comment:
```suggestion
> You can iterate over an array with the help of ``Array.forEach()`` method as well instead of using loops. It was introduced with **ES5** version.
```